x-pack/legacy/plugins/epm/server/lib/elasticsearch/index.ts

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// Default label to be used as the use case
+import { Dataset } from '../../../common/types';
+
+const DEFAULT_LABEL = 'default';
+
+/**
+ * Creates the base name for Elasticsearch assets in the form of
+ * {type}-{label}-{package}-{datasetName}
+ */
+export function getDatasetAssetBaseName(dataset: Dataset): string {
+  return dataset.type + '-' + DEFAULT_LABEL + '-' + dataset.package + '-' + dataset.name;
```

Comment:
Some other options are

```javascript
[dataset.type, DEFAULT_LABEL, dataset.package, dataset.name].join('-')
```

or (if you destructure `dataset`)
```javascript
`${type}-${label}-${pkg}-${name}`
```

Reply:
Went with a third option:

```
  return `${dataset.type}-${DEFAULT_LABEL}-${dataset.package}-${dataset.name}`;
```

Just my personal preference ;-)

Reply:
👍 I forgot the last part which was going to say "it will be nearly identical to what's in the comment". Any way which avoids manual string concatenation is a win, IMO.